Foundation.hs

```diff
@@ -130,14 +130,16 @@ instance YesodAuth App where
     -- Override the above two destinations when a Referer: header is present
     redirectToReferer _ = True
 
-    getAuthId creds = runDB $ do
-        muser <- getBy $ UniqueUser (credsPlugin creds) (credsIdent creds)
+    getAuthId Creds{..} = runDB $ do
```

Comment:
In this case, `buildUser` was dangerous because `Foundation` doesn't limit it's exports (bit of a long story, but life is far more convenient that way). It's too generic a name to be global, IMO.